src/plugins/ui_feature/renderer.ts

```diff
@@ -144,24 +145,72 @@ export class RendererPlugin extends UIPlugin {
     const areGridLinesVisible =
       !this.getters.isDashboard() &&
       this.getters.getGridLinesVisibility(this.getters.getActiveSheetId());
-    const inset = areGridLinesVisible ? 0.1 * thinLineWidth : 0;
 
     if (areGridLinesVisible) {
       for (const box of this.boxes) {
         ctx.strokeStyle = CELL_BORDER_COLOR;
         ctx.lineWidth = thinLineWidth;
-        ctx.strokeRect(box.x + inset, box.y + inset, box.width - 2 * inset, box.height - 2 * inset);
+        ctx.strokeRect(box.x, box.y, box.width, box.height);
       }
     }
   }
 
   private drawCellBackground(renderingContext: GridRenderingContext) {
-    const { ctx } = renderingContext;
+    const { ctx, thinLineWidth } = renderingContext;
+    const areGridLinesVisible =
+      !this.getters.isDashboard() &&
+      this.getters.getGridLinesVisibility(this.getters.getActiveSheetId());
     for (const box of this.boxes) {
       let style = box.style;
       if (style.fillColor && style.fillColor !== "#ffffff") {
         ctx.fillStyle = style.fillColor || "#ffffff";
-        ctx.fillRect(box.x, box.y, box.width, box.height);
+        ctx.fillRect(
+          box.x - thinLineWidth / 2,
+          box.y - thinLineWidth / 2,
+          box.width + thinLineWidth,
+          box.height + thinLineWidth
+        );
+        if (areGridLinesVisible) {
+          const { r, g, b } = colorToRGBA(style.fillColor || "#ffffff");
+          const borderColor = `rgb(${r - 30},${g - 30},${b - 30})`; // to get a darker color
+          if (!box.isOverflow) {
+            ctx.lineWidth = thinLineWidth;
+            ctx.strokeStyle = borderColor;
+            ctx.strokeRect(box.x, box.y, box.width, box.height);
+          } else {
+            // top
+            this.drawBorder(ctx, borderColor, thinLineWidth, {
+              x1: box.x - thinLineWidth / 2,
+              y1: box.y,
+              x2: box.x + box.width + thinLineWidth / 2,
+              y2: box.y,
+            });
+            //bottom
+            this.drawBorder(ctx, borderColor, thinLineWidth, {
+              x1: box.x - thinLineWidth / 2,
+              y1: box.y + box.height,
+              x2: box.x + box.width + thinLineWidth / 2,
+              y2: box.y + box.height,
+            });
+            if (box.content?.align === "left") {
```

Comment:
I don't understand what those two if statements do. If I move their body outside, I don't see anything different.

Reply:
@LucasLefevre These two if conditions are used to only render the left/right colorful border when the cell content is overflowing and left/right aligned. If the text alignment is center, both left and right colorful borders are not rendered. 
GSheets has this kind of behavior (Excel web handles this background color pretty roughly). 
![left](https://user-images.githubusercontent.com/33364182/214594939-98dbd1f0-0e73-4c74-8031-7197f5a283d9.png)
![right](https://user-images.githubusercontent.com/33364182/214594973-eddc01e6-bfc8-4fc1-9219-84fe9907497e.png)
![center](https://user-images.githubusercontent.com/33364182/214594994-7dabf400-5520-4814-ad51-a3148fbb6d05.png)